### PR TITLE
Merge Recommendation servlet feature bug

### PR DESCRIPTION
**Pull request by ethyus**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 1]
```diff
@@ -11,47 +11,33 @@
...
+      ResponseSerializer.sendErrorJson(response, "User not logged in");
     }
 
     String queryWord = request.getParameter("queryWord").toLowerCase();
     int numOfWordsRequested = Integer.parseInt(request.getParameter("numOfWordsRequested"));
+    checkforWordRequestedBound(response, numOfWordsRequested);
+
+    // Ensures db is opened in read only to avoid data perturbation
+    String path = Paths.get("").toAbsolutePath().toString() + "/word2vec.db";
+    DB db = DBMaker.fileDB(path).readOnly().make();
```
Comment (tylerhou) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java:45]: Why was this removed from init()?

**Review by tylerhou — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java [hunk 2]
```diff
@@ -64,11 +50,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
       Map<String, String[]> jsonInfo = new HashMap<>();
       jsonInfo.put(queryWord, requestedNeighbors);
+      response.setContentType("application/json;");
       response.getWriter().println(new Gson().toJson(jsonInfo));
     } catch (NullPointerException e) {
-      Map<String, String> jsonErrorInfo;
-      jsonErrorInfo = ResponseSerializer.getErrorJson("Cannot find similar words at the moment");
-      response.getWriter().println(new Gson().toJson(jsonErrorInfo));
+      ResponseSerializer.sendErrorJson(response, "Cannot find similar words at the moment");
```
Comment (zmr) [inline src/main/java/com/google/sps/servlets/RecommendationServlet.java:62]: "at the moment" could imply if the user waits a moment, there is a chance they'll get a response with similar words. Maybe keep the response to "Cannot find similar word."

**Review by zmr — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java (hunk 2 above)
Comment (ethyus) [reply src/main/java/com/google/sps/servlets/RecommendationServlet.java:62]: done

File: src/main/java/com/google/sps/servlets/RecommendationServlet.java (hunk 1 above)
Comment (ethyus) [reply src/main/java/com/google/sps/servlets/RecommendationServlet.java:45]: creating the db instance needs to be within the same local text of opening the db file and reading it. I can't put this in the init() because after the second call to the servlet, it throws an error because the db instance was closed() and I need to make the db again.
